### PR TITLE
COMMUNITY PR #1: Arnold Palmer, Auto-collecting Slime Processors, Cyborg Booping, and Maint Abandoned Crates

### PR DESCRIPTION
**Pull request by Iamgoofball**
The following 4 people TOOK INITIATIVE and joined #coderbus on irc.rizon.net via the following link to request a feature be added:
https://client02.chat.mibbit.com/?server=irc.rizon.net&channel=%23coderbus
1. Lati
2. Geo_Jerkal
3. Tanquetsunami25
4. Cocothegog0

Anyone who joins coderbus via this link:
https://client02.chat.mibbit.com/?server=irc.rizon.net&channel=%23coderbus
are eligible for the Community PR project.
Demo pictures:
dreamseeker_2016-04-09_15-33-00
dreamseeker_2016-04-09_15-33-17
dreamseeker_2016-04-09_15-32-43
:cl: Iamgoofball, Lati, Geo_Jerkal, Tanquetsunami25, and Cocothegog0
rscadd: Lati requested that the Slime Processor will now automatically scoop up any dead slimes into itself for processing.
rscadd: Geo_Jerkal requested that you be able to mix Arnold Palmers with Tea and Lemon Juice.
rscadd: Tanquetsunami25 requested that cyborg be able to boop.
rscadd: Cocothegog0 requested that Abandoned Crates be added to maint loot spawners.
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
neat ideas :+1:

**Comment by GunHog** [comment]
All of this is good stuff.

**Comment by Iamgoofball** [comment]
added new feature

**Comment by Thunder12345** [comment]
That's four people

**Comment by PKPenguin321** [comment]
maint abandoned crates, like the ones that you have to play the guessing game with? should probably make sure they are all low tier loot ones

**Comment by Zilenan91** [comment]
Ooh, the maint abandoend crates sound cool.

**Comment by Fox-McCloud** [comment]
What really needs to be a thing is that the slime console can be upgraded so that it becomes more efficient.

I know, I know, it's a computer, so it doesn't have the same things as machine frames (AFAIK), but it would still be nice if you could upgrade the monkey efficiency of it, somehow.

**Comment by Bawhoppen** [comment]
:-1: no on the fact these are the full-blown loot crates that can give soulstones, guns, fucking katanas, and shitloads of other OP items, otherwise yes

**Comment by Bawhoppen** [comment]
also I should note that many maint lootspawners are in the middle of the maint hallways, so adding impassable crates would be extremely annoying

**Comment by Archie700** [comment]
My opinions in order:

I'll be fine with the slime processor autoprocess whether it's roundstart or done as an upgrade like autocloning.

I'm fine with this

BEEP BOOP A-OK

Ok, this one I'm not so sure. Maybe only allow low-tier abandoned crates to be in maint.

**Comment by KazeEspada** [comment]
These crates have shit loot tbh. You only have 1%chance of a katana and they explode after 10 failures. Btw i calculated, it only a 5% chance of rare loot, a 20% chance of mining items, and a 75% chancce for trash.

**Comment by Bawhoppen** [comment]
I counted and I see a 30% chance of getting something good, thats 1 in 3 crates, not to mention if you get multiple your chances increase ALOT

**Comment by coiax** [comment]
The crates are hard to open though without explosives, and most of the crew decidedly don't have access to any.

**Comment by Bawhoppen** [comment]
they can be drilled open easily, not to mention just taking them to a mining mob will crack them open (or gibtonite for that matter)

**Comment by coiax** [comment]
okay, so an assistant finds an abandoned crate in maint, and then gets a shuttle to mining, and then gets a mining mob to smash it open without dying in the process. He then gets a 30% chance of "good shit". I think that sounds fair.

**Comment by Zilenan91** [comment]
He can also blow it open with an IED which is also fine because the IED can blow up in his hand and kill him

**Comment by Iamgoofball** [comment]
blowing it open with a IED should be triggering the anti-tamper though

**Comment by coiax** [comment]
Blowing it open with explosives didn't trigger it before though. Do we want to encourage people to do the confusing mastermind puzzle?

**Comment by Bawhoppen** [comment]
also it can be easily shot open by any gun 
goof just remove it from this PR and make it a seperate one, it poses too many problems to risk losing borg boops

**Comment by coiax** [comment]
Okay, I went through and picked all of the really valuable/dangerous stuff, ignoring minor weapons like drills and batons, and the beakers.

- Hand teleporter 1%
- Stetchkin 1%
- Cursed katana 1%
- Xray injector 1%
- Regular katana 1%
- Soulstone 1%
- "Weed extract" 1% (Some sort of unused xeno item)
- Compact defib 2%
- Plasma cutter 2%
- Sonic jackhammer 2%
- Bluespace crystals 2%
- KA and Spacesuit 2%
- Camo jumpsuit 4%

So I make it about 21%, probably you've included some items I haven't.

There are about 140 loot spawns on Box station, based on the numbers given in the file, which means, roughly, you got a binomial distribution (n=140, p=2/140), which gives a 86.66% of at least one crate. On average, you're getting about 2 of these crates in maint around. So that's ISH ISH ISH, about 0.4 good items a shift.

My numbers may be messy, but I don't think it's going to be that game breaking. It just means that very occasionally there's some good loot in Maint, but not most of the time.

**Comment by Bawhoppen** [comment]
It's still a large balance concern that isn't fit for this PR, not to mention the fact that maint loot spawners are in the middle of maint hallways and this will block them

**Comment by ChangelingRain** [comment]
Maint loot spawners aren't in the middle of halls, but they ARE on top of tables, racks, lockers, and other crates and that'll look weirdass. :-1:

**Comment by coiax** [comment]
Okay, take out the abandoned crates, we can make a "box" version of an abandoned crate when we get a sprite together.

But otherwise, good.

**Comment by duncathan** [comment]
I've gotta commend you goofball for making a PR to improve our PR

**Comment by Iamgoofball** [comment]
metal PR?

**Comment by Iamgoofball** [comment]
@Bawhoppen @coiax yeah, those are supposed to trigger the anti-tamper. I'll fix it in this PR.

@ChangelingRain eh

**Comment by PKPenguin321** [comment]
remove abandoned crates from maint loot please

**Comment by Iamgoofball** [comment]
Why don't I just remove all bypasses for the mastermind game and make it harder

**Comment by Iamgoofball** [comment]
also for reference:

i'm the guy who ported abandoned crates from an old copy of gooncode to baycode where it got ported here

**Comment by PKPenguin321** [comment]
okay cool but the thing is that there is like coiax posted a 20~% chance it will give some kind of stupid rare loot thing, and that number is even higher if there is more than one crate in a round. i don't think this will be good for the round since one in 5 rounds (or more) will have stupid bullshit that enables easy validhunting like a roundstart gun/xray/high brute katana etc.

not to mention you can open these things with a mastermind solver program so it's not even difficult

**Comment by Iamgoofball** [comment]
@PKPenguin321 

>mastermind solver

I'll believe it when I see it for this. Plus, I can always up the complexity and add a trademark Goof(tm) spin to it to prevent this.

I plan to drop the chances back down to the original 3. 10 chances on try is too damn much.

**Comment by PKPenguin321** [comment]
>I'll believe it when I see it for this.

ok

**Comment by Bawhoppen** [comment]
@ChangelingRain there definitely are maint loot spawners in the middle of the maint tunnels aswell

GOOF JUST DO A SEPERATE PR

**Comment by Durkel** [comment]
I'm going to be lazy and instead request this change here.

@Iamgoofball Can you update the spy syndie bundle kit? You added the chameleon kit which has more and better options for only 4 tc.

**Comment by Iamgoofball** [comment]
i'll just put a chameleon kit inside the spy bundle

**Comment by Durkel** [comment]
Good enough :+1:

**Comment by Cheridan** [comment]
>GOOF JUST DO A SEPERATE PR 

This. Just offload the mastermind crates to another PR so I can actually get this shit in.

**Comment by PKPenguin321** [comment]
freeeeee hand teleporters, xrays, katanas, and guns, let's go boyzz

**Comment by duncathan** [comment]
the crates weren't merged

File: code/modules/food&drinks/kitchen machinery/processor.dm [hunk 1]
```diff
@@ -29,6 +29,25 @@
 	for(var/obj/item/weapon/stock_parts/manipulator/M in component_parts)
 		rating_speed = M.rating
 
+/obj/machinery/processor/process()
+	..()
+	var/mob/living/simple_animal/slime/picked_slime
+	for(var/mob/living/simple_animal/slime/slime in range(1,src))
```
Comment (PJB3005) [inline code/modules/food&drinks/kitchen machinery/processor.dm]: Use `orange` instead so you can remove the check below.

File: code/modules/food&drinks/kitchen machinery/processor.dm [hunk 2]
```diff
@@ -29,6 +29,25 @@
...
+		if(slime.loc == src)
+			return
+		if(istype(slime, /mob/living/simple_animal/slime))
```
Comment (PJB3005) [inline code/modules/food&drinks/kitchen machinery/processor.dm]: This if is redundant.

File: code/modules/food&drinks/kitchen machinery/processor.dm [hunk 3]
```diff
@@ -29,6 +29,25 @@
...
+			if(slime.stat)
+				picked_slime = slime
+				break
+	if(!picked_slime)
+		return
+	var/datum/food_processor_process/P = select_recipe(picked_slime)
+	if (!P)
+		return
+
+	src.visible_message("[picked_slime] is sucked into [src].")
```
Comment (PJB3005) [inline code/modules/food&drinks/kitchen machinery/processor.dm]: `src.`

Comment (PJB3005) [inline code/modules/food&drinks/kitchen machinery/processor.dm]: Also should use `\the [src]`

File: code/modules/food&drinks/kitchen machinery/processor.dm [hunk 4]
```diff
@@ -29,6 +29,25 @@
...
+	picked_slime.loc = src
```
Comment (PJB3005) [inline code/modules/food&drinks/kitchen machinery/processor.dm]: Should use `forceMove()`

**Comment by PJB3005** [comment]
Oh whoops I thought I was commenting on the Paradise PR instead.

Did anybody even review this either way because holy shit.

**Comment by PKPenguin321** [comment]
yeah slime processor shit was pretty busted but iirc got fixed over 1 or 2 PRs

**Pull request merged**